entry_tasks/16_Pick_movie_by _genre_or_actor/by_genre_and_cast.py

```diff
@@ -0,0 +1,119 @@
+        print("You've entered wrong movie name. Please enter correct one.")
+
+
+
+def search_by_cast():
+    input_values = ['y', 'n']
+    is_search_by_cast = None
+    while is_search_by_cast not in input_values:
+        cast = input("Search by cast: ").strip()
+        if cast == "y":
+            print(f'{">>> Available Actors: "} {cast_values}')
+            while True:
+                actor = input("Enter specific actor: ").strip()
+                if actor.title() in cast_values:
+                    break
+                print("You've entered wrong actor name. Please enter correct one.")
+            list_of_movies = [key
+                              for key, list_of_values in CAST.items()
+                              if actor.title() in list_of_values]
+            if list_of_movies:
+                print(f'>>> Available movies: {list_of_movies} with {actor.title()}.')
+            else:
+                print('Value does not exist in the dictionary')
+            while True:
+                movie_list = []
+                movie = input("Enter movie: ").strip()
+                if movie.lower() in movie_dict.keys():
+                    movies = movie_dict[movie.lower()]
+                    movie_list.append(movies)
+                    print(f' >>> Movie to watch: {movie_dict[movie.lower()]}. Starring: {actor.title()}.')
+                    break
+                print("You've entered wrong movie name. Please enter correct one.")
+            break
+        if cast == "n":
+            print("Searching completed.")
+            break
+        else:
+            print("Enter 'y' or 'n' for searching by cast!")
+
+
+GENRES = {
+    "comedy": ["Meet the Parents", "Anger Management"],
+    "adventures": ["Mummy"],
+    "romantic": ["Vanilla Sky", "Meet Joe Black"],
+    "drama": ["Meet Joe Black"],
+    "thriller": ["Vanilla Sky"],
+    "action": ["Mission Impossible"],
+}
+CAST = {
+    'Meet the Parents': ['Robert De Niro', 'Ben Stiller'],
+    'Anger Management': ['Adam Sandler', 'Jack Nicholson'],
+    'Mummy': ['Brendan Fraser', 'Rachel Weisz'],
+    'Vanilla Sky': ['Tom Cruise', 'Penelope Cruz', 'Cameron Diaz'],
+    'Meet Joe Black': ['Brad Pitt', 'Anthony Hopkins'],
+    'Mission Impossible': ['Tom Cruise', 'Jeremy Renner']
+}
+if __name__ == "__main__":
+    cast_values = CAST['Meet the Parents'] + CAST['Anger Management'] + CAST['Mummy'] + CAST['Vanilla Sky'] + CAST[
+        'Meet Joe Black'] + CAST['Mission Impossible']
+    cast_values = unique_cast(cast_values)
+    genres_values = GENRES['comedy'] + GENRES['adventures'] + GENRES['romantic'] + GENRES[
+        'drama'] + GENRES['thriller'] + GENRES['action']
+    unique_list = unique_movies(genres_values)
+    # print(genres_values)
+    lowercase_list = [str(x.lower()) for x in genres_values]
+    # print(lowercase_list)
+    movie_dict = {}
+    for key in lowercase_list:
+        for value in genres_values:
+            movie_dict[key] = value
```

Comment:
why just not collect this movie dict when you do:
`lowercase_list = [str(x.lower()) for x in genres_values]`

like:
```py
for movie in genres_values:
    key = movie.lower()
    movie_dict[key] = movie
```